includes/Compatibility.php

```diff
@@ -0,0 +1,431 @@
+<?php
+/**
+ * Class Compatibility
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+use WP_Error;
+
+/**
+ * Class Compatibility
+ *
+ * @package Google\Web_Stories
+ */
+class Compatibility {
+
+	/**
+	 * WP_Error object passed back.
+	 *
+	 * @var WP_Error
+	 */
+	protected $error;
+
+	/**
+	 * WordPress version.
+	 *
+	 * @var string
+	 */
+	protected $wp_version;
+
+	/**
+	 * PHP version.
+	 *
+	 * @var string
+	 */
+	protected $php_version;
+
+	/**
+	 * Array of extensions.
+	 *
+	 * @var array
+	 */
+	protected $extensions;
+
+	/**
+	 * Array of required files.
+	 *
+	 * @var array
+	 */
+	protected $required_files = [];
+
+	/**
+	 * Class name.
+	 *
+	 * @var string
+	 */
+	protected $class_name;
```

Comment:
We could make this an array as well, check to see if AMP php classes exist. Thoughts?

Reply:
If the autoloaders exist, these classes exist.

As mentioned in other comments, no need for the `class_name` checks anymore

Reply:
Fixed in https://github.com/google/web-stories-wp/pull/4930/commits/1dbfd13bbe3419e0d98d1e5bca32010c6dfb8f43